### DISCUSSION TITLE
Fix comparison in `BaseHtml::renderSelectOptions()`

### DISCUSSION THREAD

**Discussion by WinterSilence** [yiisoft/yii2#19580] [Polls]
Current behavior of [BaseHtml::renderSelectOptions()](https://github.com/yiisoft/yii2/tree/master/framewor/helpers/BaseHtml.php) is incorrect:
- multiple selection: method always converts `$selection` and `$key` to strings
- single selection: method compare `$selection` and `$key` as strings at strict comparation
- error at non-strict comparation integer/float(`$key`) with object(`$selection`)

As result, in some cases method return invalid value or throw error.

I'm created 2 PR's:
1. #19577 Complete, but breaks BC
2. #19579 Fix only error at non-stict comparation

**Comment by papppeter** [comment]
string keys would be perfect for me!
